### PR TITLE
For Code Review

### PR DESCRIPTION
**Pull request by jayvarner**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# GeoServer Workflow
+This script automates the process for preparing GeoTIFFs for use in WMS applications and adding them to GeoServer.
```
Comment (rlskoeser) [inline README.md:2]: Is it worth adding links (or versions, or anything else?) to geotiff or geoserver - or are those things that will be obvious to anyone using this?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+## config.yaml
+Provided is a sample config file (<code>config.yaml.dst</code>). Rename that to <code>config.yaml</code> and fill in the various items.
+
+## What the script does
+* Collects all the tiff files in the configured directory
+* Runs each tiff though the following process one at a time
+	* Uses GDAL commands to prepare for WMS applications
+	* Finds the proper metadata file based on configured directory and matching file name. For example, if the script finds Sheet4.tif, it will look for Sheet4.xml
+	* Checks the metadata file for an ARK. If no ARK is found, it creates one and adds it to the metadata file
```
Comment (rlskoeser) [inline README.md:25]: Perhaps worth a brief description or link to explain an ARK? Also, do you need to document where the ARK is expected in the metadata?

File: tada.rb [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+
+  def create_ark()
+    # Private method to create an ARK via the pidman REST API.
+    pidman_auth = {
+      username: $config['pidman_user'],
+      password: $config['pidman_pass']
+    }
+
+    response = HTTParty.post \
+      'https://testpid.library.emory.edu/ark/', \
```
Comment (rlskoeser) [inline tada.rb:105]: Should be using the pidman url from the config file here, right?

File: tada.rb [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+    # Return a hash for authenticaing to the REST API.
+    auth = {
+      username: $config['geoserver_user'],
+      password: $config['geoserver_pass']
+    }
+    return auth
+  end
+end
+
+def add_store(map)
```
Comment (rlskoeser) [inline tada.rb:144]: Would it make more sense for the this function as well as the add/update layer ones to be methods on the GeoServer class?

File: README.md (hunk 2 above)
Comment (jayvarner) [inline README.md:25]: Good idea, I will add that as a TODO (still waiting on the spec, but will need the reminder).

File: README.md (hunk 1 above)
Comment (jayvarner) [inline README.md:2]: I should list the GeoServer API versions this is known to work with, thanks.

File: tada.rb [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+  end
+end
+
+def process_files(map)
+  # Method to prep GeoTIFFs for use in WMS applications.
+  in_dir = $config['in_dir']
+  tmp_dir = $config['tmp_dir']
+  out_dir = $config['out_dir']
+
+  warp = "gdalwarp -s_srs EPSG:2240 -t_srs EPSG:4326 -r average\
```
Comment (rlskoeser) [inline tada.rb:240]: Do you need any error handling if these gdal commands are not installed?

File: README.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+## Dependencies
+You will need ruby and a few gems:
+
+<code>gem install net-sftp</code>
+
+<code>gem install httparty</code>
+
+<code>gem install nokogiri</code>
+
```
Comment (rlskoeser) [inline README.md:16]: Perhaps also mention the gdal commands that need to be installed?

File: README.md (hunk 1 above)
Comment (rlskoeser) [inline README.md:2]: I don't actually know what WMS applications are either

**Comment by rlskoeser** [comment]
I like that you've provided such a thorough readme, and helpful comments throughout so it's fairly easy to track what's going even without a lot of familiarity with the stuff you're working with.

It looked like there are still a lot of todos, are any of those done, or are they still in process?

Still seems weird to have a whole git repo for a single script (although I know we've disagreed on that before, too :-) ).  Is this something that you'll have tagged releases of, and that people will install independently of the other codebases for the project?  I would probably have stuck it in a scripts directory or subdirectory of the related application (with readme and config files in that directory).  Or if it's truly a standalone item, perhaps a github gist with documentation via inline comments would be sufficient.

File: tada.rb (hunk 3 above)
Comment (jayvarner) [inline tada.rb:105]: ha, yes, good catch.

File: README.md (hunk 1 above)
Comment (jayvarner) [inline README.md:2]: @rlskoeser: Web Mapping Service

File: tada.rb (hunk 4 above)
Comment (jayvarner) [inline tada.rb:144]: @rlskoeser I went back and forth on this but decided to go ahead a do it. I think it does make things a little cleaner. Thanks.

File: tada.rb (hunk 5 above)
Comment (jayvarner) [inline tada.rb:240]: Yes, and added.

**Comment by jayvarner** [comment]
@rlskoeser Thanks for looking this over and making suggestions. I think I've addressed most, if not all, of them.

**Comment by rlskoeser** [comment]
Looks good!

**Pull request merged**